core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -86,6 +92,17 @@
         this.scheme = scheme;
         inboundTrafficController = InboundTrafficController.ofHttp1(channel);
         this.encoder = encoder;
+        encoder.webSocketUpgradeContext().setWebSocketUpgradeListener(this);
+    }
+
+    @Override
+    public void upgraded(boolean success) {
+        if (!success) {
+            if (req instanceof DefaultDecodedHttpRequest) {
+                req.close();
+            }
```

Comment:
Could you explain why we call `.close()` only on a `DefaultDecodedHttpRequest`? Please add comments.